docs/tasks/run-as-deployment/quickdockerbuildusingidentitykeyvault/azuredeploy.json

```diff
@@ -0,0 +1,126 @@
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "registryName": {
+      "type": "string",
+      "minLength": 5,
+      "maxLength": 50,
+      "metadata": {
+        "description": "Name of your Azure Container Registry"
+      }
+    },
+    "registrySku": {
+      "type": "string",
+      "metadata": {
+        "description": "Tier of your Azure Container Registry"
+      },
+      "defaultValue": "Basic",
+      "allowedValues": [
+        "Basic",
+        "Standard",
+        "Premium"
+      ]
+    },
+    "registryAdminUserEnabled": {
+      "type": "bool",
+      "defaultValue": false,
+      "metadata": {
+        "description": "Enable admin user that have push / pull permission to the registry"
+      }
+    },
+    "taskRunName": {
+      "type": "string",
+      "minLength": 5,
+      "maxLength": 50,
+      "metadata": {
+        "description": "Name of your Task Run"
+      }
+	},
+	"managedIdentityName": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 50,
+      "metadata": {
+        "description": "Name of your ManagedIdentity"
+      }
+    }
+  },
+ "variables": {
+	"repository": "hello-world-node",
+	"tag": "parameters('taskRunName')",
+	"imageName": "[concat(variables('repository'), ':', parameters('taskRunName'))]",
+	"idApiVersion": "[first(providers('Microsoft.ManagedIdentity', 'userAssignedIdentities').apiVersions)]",
+	"managedIdResourceId": "[resourceId('f9d7ebed-adbd-4cb4-b973-aaf82c136138', 'huanwudfwestgroup', 'Microsoft.ManagedIdentity/userAssignedIdentities', parameters('managedIdentityName'))]"
+},
+  "resources": [
+    {
+      "type": "Microsoft.ContainerRegistry/registries/taskRuns/",
+      "name": "[concat(parameters('registryName'), '/', parameters('taskRunName'))]",
+      "location": "[parameters('location')]",
+      "apiVersion": "2019-06-01-preview",
+      "identity": {
+        "principalId": null,
+        "tenantId": null,
+        "type": "UserAssigned",
+        "userAssignedIdentities": {
+            "[variables('managedIdResourceId')]": {
+            }
+        }
+	  },
+      "properties": {
+          "runRequest": {
+            "type": "DockerBuildRequest",
+            "imageNames": [
+                "[variables('imageName')]"
+            ],
+            "sourceLocation": "https://github.com/huanwu/armtemplates.git",
+            "dockerFilePath": "Dockerfile-test2",
+            "values": [],
+            "isPushEnabled": true,
+            "platform": {
+                "os": "linux",
+                "architecture": "amd64"
+            },
+            "credentials": {
+                "apiVersion": "2018-09-01",
+                "customRegistries": {
+                    "huanglitest05.azurecr-test.io": 
```

Comment:
Can the registry and following vault secret value be passed from parameters?

Reply:
Yes, I can add a parameter on it. But you can only pass one.

Reply:
I will pass in the registry name and keyvault name.